### ISSUE TITLE
Pokemon Tower Spinner Soft Lock

### ISSUE THREAD

**Issue by OmniM00se** [Dabomstew/pokered-speedchoice#22]
TLDR; Do NOT hug the wall on the top floor of Pokemon Tower if you have Spinner set to Hell or Why. Else, you risk soft locking your game after you battle and defeat a Rocket who sees you and walks up or down to initiate battle (obviously, blacking out during the fight would be a way to avoid soft locking, if necessary).

Original Post (Edited for Clarity): 
If Spinner is set to Hell or Why, the player can soft lock on the top floor of Pokemon Tower after fighting a Rocket. I just played a Hell game where my character encountered the right-side Rocket against the right wall, and after the fight was over, he walked a seemingly random path around the lower-right quadrant of the screen until he disappeared. I was playing a Nuzlocke and had no recent saves, so I waited and fast forwarded for quite a long time in hopes he might reach his expected destination coordinates and that would allow my character to move, but to no avail. No video of this incident unfortunately, and I've played spinner Hell without this issue before, which suggests to me that being against the wall and/or talking to the Rocket directly (I don't remember if I did or not) could have been a factor in corrupting his walk path.

Edited to Add: I retested this in the same ROM and generally, there's a risk of soft lock if the player is directly below or above any Rocket when they're seen (changing the Rocket's Y coordinate appears to be the cause). For 1 tile away encounters, I tested both talking to the Rocket and allowing myself to be seen, but results were the same.

-Rocket 1 (Left): Engaging from the single tile below, or any # of tiles to the right is safe. Engaging from any number of tiles above will result in soft lock.
-Rocket 2 (Right): Engaging from 1 tile below, or any # of tiles to the left is safe. Engaging from >1 tile below, or any number of tiles above will result in soft lock.
-Rocket 3 (Left): Engaging from 1 tile below or above, or any # of tiles to the right is safe (if engaged from above, the Rocket's walk path is longer, but still completes successfully. Maybe don't risk it!). Engaging from >1 tile below or above results in soft lock.
